src/Servers/Kestrel/shared/TransportConnection.cs

```diff
@@ -65,6 +65,8 @@ public override string ConnectionId
             }
         }
 
+        public virtual Socket Socket { get; } = default!;
```

Comment:
```suggestion
        public Socket? Socket { get; }
```

Reply:
This should be virtual I guess because we override in `SocketConnection.cs` and sets Socket!

Reply:
I should have added a protected setter. I don't see the point in making this virtual since the override is basically the same.